algorithms/src/Kokkos_Random.hpp

```diff
@@ -901,52 +901,51 @@ class Random_XorShift64_Pool {
   KOKKOS_DEFAULTED_FUNCTION Random_XorShift64_Pool& operator=(
       Random_XorShift64_Pool const&) = default;
 #else
-  Random_XorShift64_Pool()   = default;
+  Random_XorShift64_Pool() = default;
 #endif
+
   Random_XorShift64_Pool(uint64_t seed) {
     num_states_ = 0;
 
     init(seed, execution_space().concurrency());
   }
 
+  Random_XorShift64_Pool(execution_space exec, uint64_t seed) {
```

Comment:
For consistency with similar functionalities in the Kokkos API and to avoid unnecessarily incrementing the reference counter.
```suggestion 
  Random_XorShift64_Pool(execution_space const& exec, uint64_t seed) {
```